### ISSUE TITLE
on-top demo draws over application

### ISSUE THREAD

**Issue by KnockerPulsar** [david-vanderson/dvui#40]
Hello, thanks so far for your work on this library, quite cool to see something like this written for zig! Also, sorry if this issue is more fit for the demo's repo, I'm not quite sure if it's an issue with the demo itself or the library.

Anyway, I seem to have some trouble rendering dvui on top of my existing application. Here's some context: I draw pixels to some buffer, then I draw this buffer onto the screen. At first I used minifb, but it seems to not work well with dvui, so I transitioned to SDL. 

I'm pretty much using the on-top demo, but adding a bit of application specific initialization and the following block inside the rendering part of the main loop:
```zig
  // Somehwere before the main loop:
  var texture = c.SDL_CreateTexture(window.renderer, c.SDL_PIXELFORMAT_ARGB8888, c.SDL_TEXTUREACCESS_STREAMING, width, height);

  // Somewhere inside the main loop
        window.backend.clear();
        try dvui_stuff();

        // Request from SDL to map the texture to some memory location so we can write to it
        var pixels: ?*anyopaque = null;
        var pitch: c_int = undefined;
        _ = c.SDL_LockTexture(texture, null, &pixels, &pitch);

        var pixelsAsU32: [*]u32 = @alignCast(@ptrCast(pixels));

        // For each pixel, copy over some data (in this case, just display bright purple)
        for (pixels) |pixel| {
                const x = pixel.x;
                const y = pixel.y;
                const index = x + (height - y - 1) * width;  // Defined somewhere else in the program
                pixelsAsU32[index] = RGBA(1, 0, 1, 1); // Convert RGBA to a u32        
        }

        c.SDL_UnlockTexture(texture);

        // Render the whole texture onto the whole screen.
        _ = c.SDL_RenderCopy(window.renderer, texture, null, null);

        // marks end of dvui frame, don't call dvui functions after this
        // - sends all dvui stuff to backend for rendering, must be called before renderPresent()
        _ = try win.end(.{});
```

Running the program in renderdoc, it shows that the texture was properly uploaded to the GPU (although flipped), when I do anything with dvui, the call to draw my texture draws only a part of it.  If I don't call `dvui_stuff()`, the texture is rendered fully. 

When observing the calls in renderdoc, specifically the call to draw my texture, there seems to be a `glEnable(GL_SCISSORS_TEST)` that doesn't appear when `dvui_stuff()` is commented, but I'm not familiar with OpenGL's scissor feature much, so I'm not sure if that's the root cause of the issue. It's very suspicious though. 

When running the application, it shows the texture I'm drawing and dvui for the first frame only, then only dvui from then on out with a black background. If I move the dvui floating window, I can see "ghosts" of my texture, but they seem to follow the floating windows and disappear behind them.

At this point, I'm not sure if it's something in my code or perhaps with dvui.

**Comment by david-vanderson** [comment]
Thanks very much for the detailed report!

It was definitely a bug in dvui itself.  The SDLBackend was not properly handling the clipping (gl scissor) region.  It was super useful that you mentioned the scissor difference.

Because dvui was not saving/restoring the clipping, after the first frame you were stuck with whatever the last clipping region dvui happened to use.  After fixing that, there was the twist that if scissor is not enabled and you get the current scissor region, you get an empty rectangle!  So additionally dvui now sets the scissor region to the whole screen during begin().

I've committed a fix for it.  Does that fix the problem for you?

**Comment by KnockerPulsar** [comment]
> Thanks very much for the detailed report!
> 
> It was definitely a bug in dvui itself. The SDLBackend was not properly handling the clipping (gl scissor) region. It was super useful that you mentioned the scissor difference.

Thank **you** for the quick followup! I suspected it was the scissor call but didn't really know how things work well enough to fix it myself.

> I've committed a fix for it. Does that fix the problem for you?

Yep, seems to fix things alright. 

Note: I did struggle a bit to make zig package manager pull the latest version, especially that the on-top example suggests using dvui as a submodule. I tried making that work for a bit but the mixing between vendored code (dvui) and automatic dependency management (freetype) was too much for my zig expertise. 

In the end I resorted to manually changing the URL in `build.zig.zon` since the last part looked like a commit hash. I was too lazy to lookup how to get the hash, so I ran the build step with URL only. Luckily, it printed the hash line exactly so I copied it over to the `build.zig.zon` and things worked magically from there. I do suggest updating the examples with some pointers to make things smoother for other users.
